### PR TITLE
Fix auth handling in Spring Cloud Config Client

### LINKED ISSUE #10350: [spring-cloud-config-client] Authentication not added when spring-cloud-config.url defines a endpoint, not a base url
**Description**
When defining a config with a full endpoint url like https://my-domain-url.com/config
Quarkus does not succeed in getting configuration when it's protected by Basic Authentication 

**Expected behavior**
Successful config retrieve

**Actual behavior** : 401 http response
```
Caused by: java.lang.RuntimeException: Got unexpected HTTP response code 401 from https://my-domain-url.com/config/config/my-app-name/dev
        at io.quarkus.spring.cloud.config.client.runtime.DefaultSpringCloudConfigClientGateway.exchange(DefaultSpringCloudConfigClientGateway.java:66)
        at io.quarkus.spring.cloud.config.client.runtime.SpringCloudConfigServerClientConfigSourceProvider.getConfigSources(SpringCloudConfigServerClientConfigSourceProvider.java:37)
```

**Configuration**
```properties
quarkus.application.name=my-app-name
quarkus.spring-cloud-config.username=user
quarkus.spring-cloud-config.password=pwd
quarkus.spring-cloud-config.fail-fast=true
quarkus.spring-cloud-config.enabled=true
quarkus.spring-cloud-config.url=https://my-domain-url.com/config
```
**Environment :**
 - java version: 1.8.0_181 
 - Quarkus version: 1.4.2 (tried alson in 1.5.2)
 - Build tool : Gradle , version 6.4.1

**Step by step :**
To help resolve the issue , at start-up : 
1) DefaultSpringCloudConfigClientGateway#determineBaseUri line 40 : 
baseUri is the full endpoint "https://my-domain-url.com/config" instead of base uri "https://my-domain-url.com"

2) DefaultSpringCloudConfigClientGateway#setupContext line 83 
``` 
 authCache.put(HttpHost.create(baseUri), new BasicScheme());
 ```
 3)  org.apache.http.client.protocol.RequestAuthCache#process line 53
 Target host is https://my-domain-url.com, port is < 0
 
line 58 :
 ```
 target = new HttpHost(target.getHostName(), route.getTargetHost().getPort(), target.getSchemeName());
```
At this point target is https://my-domain-url.com:443
line 63 : 
```
 AuthScheme authScheme = authCache.get(target);
```
Result : AuthScheme is null, and function RequestAuthCache#doPreemptiveAuth is never call to set Authentication Basic

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #10350

### CODE REVIEW

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java [hunk 1]
```diff
@@ -30,57 +34,74 @@
...
-    private final String baseUri;
+    private final URI baseURI;
 
     public DefaultSpringCloudConfigClientGateway(SpringCloudConfigClientConfig springCloudConfigClientConfig) {
         this.springCloudConfigClientConfig = springCloudConfigClientConfig;
-        this.baseUri = determineBaseUri(springCloudConfigClientConfig);
+        try {
+            this.baseURI = determineBaseUri(springCloudConfigClientConfig);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```
Comment (gsmet) [inline extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java:44]: Maybe we could have a message here?

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java [hunk 2]
```diff
@@ -30,57 +34,74 @@
...
     }
 
-    private String determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClientConfig) {
-        String baseUri = springCloudConfigClientConfig.url;
-        if (null == baseUri || baseUri.isEmpty()) {
-            throw new IllegalArgumentException("baseUri cannot be empty");
+    private URI determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClientConfig) throws URISyntaxException {
+        String url = springCloudConfigClientConfig.url;
+        if (null == url || url.isEmpty()) {
+            throw new IllegalArgumentException("baseUrl cannot be empty");
```
Comment (gsmet) [inline extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java:51]: I would make the message more explicit by adding some context (typically which extension or the full property name)?

**Review by gsmet — APPROVED** [review]
Makes sense. I added some minor comments, one not directly related to this PR but I thought we could improve things a bit while at it. Feel free to ignore them if it's too much nitpicking :).

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java (hunk 2 above)
Comment (geoand) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java]: Right, that makes sense

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/DefaultSpringCloudConfigClientGateway.java]: Sure

**Review by gastaldi — APPROVED** [review]

**Pull request merged**